lib/setup.py

```diff
@@ -35,8 +35,9 @@
     "click>=7.0, <9",
     "numpy>=1.20, <3",
     "packaging>=20, <25",
-    # Lowest version with available wheel for 3.7 + amd64 + linux
-    "pandas>=1.3.0, <3",
+    # Pandas <1.4 has a bug related to deleting columns in a DataFrame changing
+    # the index dtype.
+    "pandas>=1.4.0, <3",
```

Comment:
Pandas 1.4 was released in January 2022. Download statistics show that `1.3` still has ~40 million download from a total of 250 million downloads. So dropping it is not insignificant, but we don't have any specific information that people need 1.3 support in the context of Streamlit. Might be good to merge, and if there are major complains roll this back and implement a dedicated solution for Pandas < 1.4.